src/fhirtypes/StructureDefinition.ts

```diff
@@ -494,6 +494,18 @@ export class StructureDefinition {
         }
       }
 
+      // If the element is an extension, and we found that extension via some other identifier than the sliceName
+      // we want to replace the path to use the sliceName, since we can assume that was the user's intent
+      if (
+        pathPart.base === 'extension' &&
```

Comment:
This `if` is a part over each `.` separated part of the path. So on each iteration of the loop, `pathPart.base` represents that part of the path. So if the extension is nested, this check should still work fine, because when we get to the nested extension, `pathPart.base` will just be `extension`.

I think yes, it is possible that this could happen on `modifierExtension`, I will add that.

Reply:
Done in https://github.com/FHIR/sushi/pull/772/commits/cbfb200312b54f54161ee000ac050c1c964d32db.

Reply:
Ah, right, that makes sense to me!